src/exe.mli

```diff
@@ -42,13 +42,15 @@ val build_and_link
   :  program:Program.t
   -> linkages:Linkage.t list
   -> ?link_flags:(unit, string list) Build.t
+  -> ?package:Package.t
   -> Compilation_context.t
   -> unit
 
 val build_and_link_many
   :  programs:Program.t list
   -> linkages:Linkage.t list
   -> ?link_flags:(unit, string list) Build.t
+  -> ?package:Package.t
```

Comment:
Are we passing this parameter anywhere? I can't see where we are.

Also, perhaps you should handle it in the compilation context. I believe it will be easier for us to get this high level parameter right if we handle at a higher level. If we have to make a choice for every executable, we are less likely to get consistent results.

Reply:
We use it in `Link_time_code_gen` to determine the main version number. Passing it through Compilation_context seems good indeed, I'm doing that